### ISSUE TITLE
Unexpected backslash ( \ ) when opening git bash with oh my zsh in vscode terminal

### ISSUE THREAD

**Issue by prime0x2** [microsoft/vscode#212227]
### Describe the bug

I have git in my Windows 10. I installed zsh and was trying to add oh my zsh to give it new look.
Everything is fine except when I open Git Bash in terminal it adds a backslash before my prompt. I've added a screenshot below for reference. 

![image](https://github.com/ohmyzsh/ohmyzsh/assets/48878200/8d10a3a4-1c61-4b0f-8f04-7f0613efc0bb)

Note: This is happening on my office laptop (windows 10). It also happened when I tried to do the same thing on my personal laptop (windows 10). Same scenario, same issue. This only happens when I open Git Bash in vscode terminal. It doesn't happen when I open Git Bash directly. Here is a screenshot of Git Bash opening directly

![image](https://github.com/ohmyzsh/ohmyzsh/assets/48878200/b10aca0c-307a-4fbc-8887-07a58f13a4ab)

This only happens in Git Bash. I installed WSL Ubuntu and did the same thing, but there is no problem. When I open WSL ubuntu in vscode terminal, there is no backslash (as it should be).

This was happening for a long time on my personal laptop. Tried to find any solution, but didn't find anything regarding this issue so I thought may be it's my machine. But I got a new job and got a new laptop, same problem appears. That's when I knew something is off.

So, I was hoping is there any solution to this?

I have also created an issue to https://github.com/ohmyzsh/ohmyzsh/issues/12398  here. But they are saying its an issue with VS Code version. https://github.com/ohmyzsh/ohmyzsh/issues/11765 check here

### Steps to reproduce

OS: Windows 10

1. Install Git Bash to the system
2. Install ZSH, VS Code
3. Install Oh My ZSH
4. Open VS Code terminal (By default it opens powershell)
5. From VS Code terminal list, open Git Bash

### Expected behavior

This is WSL when opening in VS Code terminal

![image](https://github.com/ohmyzsh/ohmyzsh/assets/48878200/23a57af7-1dce-48c8-847b-699759bdba8f)

This is when I open Git Bash directly

![image](https://github.com/ohmyzsh/ohmyzsh/assets/48878200/b10aca0c-307a-4fbc-8887-07a58f13a4ab)

This is how it should be

### Screenshots and recordings

This is when I open Git Bash from VS Code terminal

![image](https://github.com/ohmyzsh/ohmyzsh/assets/48878200/8d10a3a4-1c61-4b0f-8f04-7f0613efc0bb)

### OS / Linux distribution

Windows 10

### Zsh version

5.9

### Oh My Zsh version

master (668ca3a)

### Terminal emulator

Git Bash

Does this issue occur when all extensions are disabled?: Yes/No
Yes

- VS Code Version: 
Version: 1.89.0 (user setup)
Commit: b58957e67ee1e712cebf466b995adf4c5307b2bd
Date: 2024-05-01T02:09:22.859Z
Electron: 28.2.8
ElectronBuildId: 27744544
Chromium: 120.0.6099.291
Node.js: 18.18.2
V8: 12.0.267.19-electron.0
OS: Windows_NT x64 10.0.19045
- OS Version: Windows 10

**Comment by mcornella** [comment]
This was seen to be fixed in the latest insider version (https://github.com/ohmyzsh/ohmyzsh/issues/11765#issuecomment-2081124930) but it seems you still see this issue in the latest version so that's weird... 

The minimal code that was seen to reproduce the issue (print a backslash character) was:

```sh
printf '\e]7;file://%s%s\e\' HOSTNAME /c/Users/user/Desktop
```

In theory this string is following the OSC7 standard sequence.

**Comment by prime0x2** [comment]
> This was seen to be fixed in the latest insider version ([ohmyzsh/ohmyzsh#11765 (comment)](https://github.com/ohmyzsh/ohmyzsh/issues/11765#issuecomment-2081124930)) but it seems you still see this issue in the latest version so that's weird...
> 
> The minimal code that was seen to reproduce the issue (print a backslash character) was:
> 
> ```shell
> printf '\e]7;file://%s%s\e\' HOSTNAME /c/Users/user/Desktop
> ```
> 
> In theory this string is following the OSC7 standard sequence.

I followed this ( https://github.com/ohmyzsh/ohmyzsh/issues/11765#issuecomment-2081124930 ) but it still happens. Not fixed yet.

Note: I'm not using MSYS. I'm using Git Bash.

**Comment by meganrogge** [comment]
does disabling shell integration, `terminal.integrated.shellIntegration.enabled`, help?

**Comment by prime0x2** [comment]
> does disabling shell integration, `terminal.integrated.shellIntegration.enabled`, help?

no. already tried that but no help

**Comment by meganrogge** [comment]
did this reproduce in earlier versions of vs code?

**Comment by prime0x2** [comment]
yeah, I had this problem for around 7/8 months. At that time, I only had one machine, so I always thought it was my problem alone. Then recently when I got new machine from office, I faced same issue. That's when I decided to open ticket.

**Comment by kirito41dd** [comment]
same problem
![图片](https://github.com/user-attachments/assets/b28c4695-06b4-4571-b57b-b731e5d9aae2)

**Comment by Tyriar** [comment]
@prime0x2 on [Insiders](https://code.visualstudio.com/insiders/) does it reproduce if you set `"terminal.integrated.experimental.windowsUseConptyDll": true`?
